lib/elixir/lib/kernel.ex

```diff
@@ -2186,7 +2186,12 @@ defmodule Kernel do
           _          -> arg
         end
 
-        quote do: Module.put_attribute(__MODULE__, unquote(name), unquote(arg))
+        cond do
```

Comment:
Since we already check for those in the function case clause above, I would unify both.